### PR TITLE
auto_route implementation

### PR DESCRIPTION
**Pull request by ghedwards**

### CODE REVIEW

**Comment by ghedwards** [comment]
#43

**Comment by ghedwards** [comment]
Is this how I link this to the Issue ?

**Comment by ghedwards** [comment]
I'm new to flutter, how do I fix the tests ?

**Comment by VictorUvarov** [comment]
Looks like the failing tests come from the two conflicting error members in the view model

**Comment by VictorUvarov** [comment]
Also looks like auto_route broke on flutter version 1.20.0, which makes me think the dependency might not be worth while, if it breaks on every new flutter version.

**Comment by VictorUvarov** [comment]
I closed the other PRs. I had to fix the github action test. So i just went ahead and updated the dependencies and fixed the errors. Feel free to rebase and continue working on adding `auto_route`.

File: lib/main.dart [hunk 1]
```diff
@@ -41,7 +41,7 @@ class MyApp extends StatelessWidget {
           debugShowCheckedModeBanner: false,
           material: (_, __) => MaterialAppData(
             theme: themes.primaryMaterialTheme,
-            darkTheme: themes.darkMaterialTheme,
+            // darkTheme: themes.darkMaterialTheme,
```
Comment (VictorUvarov) [inline lib/main.dart]: Why remove dark theme?

File: lib/ui/router.dart [hunk 2]
```diff
@@ -1,52 +1,13 @@
...
-    }
-  }
-
-  // Add routes that should behave as fullScreenDialogs
-  static final _fullScreenDialogs = [
-    // Routes.route_1,
-    // Routes.route_2,
-  ];
-}
+@MaterialAutoRouter(routes: [
```
Comment (VictorUvarov) [inline lib/ui/router.dart]: Please use adaptive routes for material and Cupertino routes

File: lib/ui/router.dart [hunk 3]
```diff
@@ -1,52 +1,13 @@
...
+  MaterialRoute(page: MainView),
+  MaterialRoute(page: LoginView),
+  MaterialRoute(page: StartUpView, initial: true),
+  MaterialRoute(page: PostDetailsView)
```
Comment (VictorUvarov) [inline lib/ui/router.dart]: same here

File: lib/ui/widgets/stateful/post_tile/post_tile_view_model.dart [hunk 4]
```diff
@@ -15,6 +15,7 @@ class PostTileViewModel extends BaseViewModel {
   }
 
   void showPostDetails() {
-    _navigationService.pushNamed(ViewRoutes.post_details, arguments: _post);
+    _navigationService.pushNamed(Routes.postDetailsView,
+        arguments: PostDetailsViewArguments(post: _post));
```
Comment (VictorUvarov) [inline lib/ui/widgets/stateful/post_tile/post_tile_view_model.dart:19]: Add trailing comma here

File: lib/main.dart (hunk 1 above)
Comment (ghedwards) [reply lib/main.dart]: Wasn't intentional, was doing some testing

File: test/ui/widgets/post_tile/post_tile_view_model_test.dart [hunk 5]
```diff
@@ -44,17 +44,17 @@ void main() {
...
 
         // act
         postTileViewModel.showPostDetails();
 
         // assert
         verify(mockNavigationService.pushNamed(
-          ViewRoutes.post_details,
-          arguments: mockPost1,
+          Routes.postDetailsView,
+          arguments: isA<PostDetailsViewArguments>(),
```
Comment (VictorUvarov) [inline test/ui/widgets/post_tile/post_tile_view_model_test.dart:57]: Shouldn't this line be 
```dart
PostDetailsViewArguments(post: mockPost1)
```
?

**Review by VictorUvarov — CHANGES_REQUESTED** [review]
Looks good to me. Does auto route generator run on 1.20?

- some minor changes

File: test/ui/widgets/post_tile/post_tile_view_model_test.dart (hunk 5 above)
Comment (ghedwards) [reply test/ui/widgets/post_tile/post_tile_view_model_test.dart:57]: That doesn't work, apparently isA is the right thing to do here...

File: lib/ui/router.dart (hunk 2 above)
Comment (ghedwards) [reply lib/ui/router.dart]: Will fix

File: lib/ui/router.dart (hunk 3 above)
Comment (ghedwards) [reply lib/ui/router.dart]: Will fix

File: lib/ui/widgets/stateful/post_tile/post_tile_view_model.dart (hunk 4 above)
Comment (ghedwards) [reply lib/ui/widgets/stateful/post_tile/post_tile_view_model.dart:19]: Where abouts, can you please provide ?

Comment (VictorUvarov) [reply lib/ui/widgets/stateful/post_tile/post_tile_view_model.dart:19]: after 
```dart
PostDetailsViewArguments(post: _post),
```

File: pubspec.yaml [hunk 6]
```diff
@@ -46,7 +46,12 @@ dependencies:
 
   # UI Widgets
   cupertino_icons: ^0.1.3
-  flare_flutter: ^2.0.6
+  flare_flutter: ^2.0.3
+  auto_route: ^0.6.7
+
+dependency_overrides:
```
Comment (VictorUvarov) [inline pubspec.yaml]: You can remove all these `dependency_overrides` they are not needed anymore

**Review by VictorUvarov — CHANGES_REQUESTED** [review]
Looking good, some minor fixes

File: pubspec.yaml (hunk 6 above)
Comment (ghedwards) [reply pubspec.yaml]: Done

**Review by VictorUvarov — APPROVED** [review]
Thanks for the PR!

**Pull request merged**